beacon-chain/forkchoice/protoarray/optimistic_sync.go

```diff
@@ -245,9 +232,8 @@ func (f *ForkChoice) Invalid(ctx context.Context, root [32]byte) error {
 		}
 	}
 
-	// Update the parent's best child and best descendant
-	if needsUpdateParentsChildren {
-		// look for a new child
+	// Update the parent's best child and best descendant if necessary.
+	if parent.bestChild == idx || parent.bestDescendant == idx {
```

Comment:
Are you sure this is safe? the reason I had the boolean up above is that we change the parent/child indices in the loop above this one. `idx` does not mean the same that it used to mean above.